schemas/post-artifact-request.yml

```diff
@@ -3,6 +3,121 @@ title:              "Post Artifact Request"
 type:               object
 oneOf:
+  - title: "Blob Artifact Request"
+    description: |
+      Request a list of requests in a generalized format which can be run to
+      upload an artifact to storage managed by the queue.
+    type: object
+    properties:
+      storageType:
+        description: Artifact storage type, in this case `'blob'`
+        type: string
+        enum:
+          - blob
+      expires:
+        description: |
+          Date-time after which the artifact should be deleted. Note, that
+          these will be collected over time, and artifacts may remain
+          available after expiration. S3 based artifacts are identified in
+          azure table storage and explicitly deleted on S3 after expiration.
+        type: string
+        format: date-time
+      contentType:
+        description: |
+          Artifact mime-type, when uploading artifact to the signed
+          `PUT` URL returned from this request this must given with the
+           `ContentType` header. Please, provide correct mime-type,
+           this make tooling a lot easier, specifically,
+           always using `application/json` for JSON artifacts.
+        type: string
+        maxLength: 255
+      encoding:
+        description: |
+          Optionally provide an encoding type which should be set as the HTTP
+          Content-Encoding header for this artifact.
+        type: string
+        maxLength: 255
+      sha256:
+        description: |
+          The complete SHA256 value of the entire artifact.  This must be the
+          SHA256 of the file which is to be uploaded.  For single part uploads,
+          the upload will fail if the SHA256 value of what is uploaded does not
+          match this value.  If content-encoding is to be used, this value must
+          be the overall value of the pre-encoding copy of the file.  In simple
+          terms, this is a SHA256 value of the actual thing you're wanting to
+          store.  This value must always be provided.
+        type: string
+        pattern: "$[a-fA-F0-9]{64}$"
+        minLength: 64
+        maxLength: 64
+      transferSha256:
+        description: |
+          The complete SHA256 value of the entire artifact after being encoded
+          with the content-encoding scheme.  For single part uploads, this
+          value will be the hash of the bytes transfered to S3 and for
+          multipart uploads, this value will be the hash of the overall encoded
+          file that is transfered in portions (e.g. the value of
+          X-Amz-Content-Sha256).  This value is optional when content-encoding
+          is the identity scheme (i.e. no encoding).  This value can be
+          provided with identity encoding if it's valid.  If non-identity
+          encoding is used then the value must be provided
+        type: string
+        pattern: "$[a-fA-F0-9]{64}$"
+        minLength: 64
+        maxLength: 64
+      size:
```

Comment:
I'm still not convinced we shouldn't make `transferLength` and `transferSha256` required... We can always make things optional later, we never make them required later...